core/controllers/collection_editor.py

```diff
@@ -28,6 +28,28 @@
 current_user_services = models.Registry.import_current_user_services()
 
 
+def get_matching_exploration_dicts(query_string, search_cursor):
+    """Given a query string and a search cursor, returns a list of exploration
+    dicts that satisfy the search query.
+    """
+
+    exp_ids, new_search_cursor = (
+        exp_services.get_exploration_ids_matching_query(
+            query_string, cursor=search_cursor))
+    exploration_list = []
+
+    exploration_list = (
+        summary_services.get_collection_node_exploration_dicts(
+            exp_ids))
+
+    if len(exploration_list) == feconf.DEFAULT_QUERY_LIMIT:
```

Comment:
I think we don't need to do this check + log here. If there's an issue with too few explorations coming up for a query, someone will report it and we can fix it reactively.